beamer/agent/chain.py

```diff
@@ -611,73 +615,76 @@ def maybe_prove(claim: Claim, context: Context) -> None:
     if not can_prove:
         return
 
-    if request.fill_tx is not None:
-        proof_tx = request.fill_tx
-    elif claim.invalidation_tx is not None:
-        proof_tx = claim.invalidation_tx
-
-    if proof_tx in context.l1_resolutions:
+    if proved_tx in context.l1_resolutions:
         return
 
+    assert proved_tx is not None
+
     future = context.task_pool.submit(
         run_relayer_for_tx,
         context.config.base_chain_rpc_url,
         context.target_chain.rpc_url,
         context.source_chain.rpc_url,
         context.config.account.key,
-        proof_tx,
+        proved_tx,
         True,
     )
 
     def on_future_done(f: Future) -> None:
         try:
             f.result()
         except Exception as ex:
-            context.logger.error("Optimism prove failed", ex=ex, tx_hash=proof_tx)
+            context.logger.error("Optimism prove failed", ex=ex, tx_hash=proved_tx)
         else:
-            claim.message_proved = True
+            assert request is not None
```

Comment:
Why is this assert needed? We already have that check outside of this closure.

Reply:
because mypy thinks otherwise maybe it is because this scope is different and mypy has no idea hey i already checked this

Reply:
Item "None" of "Optional[Request]" has no attribute "fill_tx"

Reply:
Ugh, mypy is not smart enough to deduce that. Oh well.